### DISCUSSION TITLE
Disassembly

### DISCUSSION THREAD

**Discussion by infromthecold** [maziac/DeZog#73] [Ideas]
Hey, this may be already available, if not, it would be a nice feature. I know you have disassembly at cursor, which i think is only working at program counter cursor, however I would like to disassembly at label or where my mouse is, or a command like -dis $address/lable, [instruction count][num bytes]

Also i would like it if disassembly would come from the next/cspect memory just like the memory viewer rather than the list file. 

Doe the debugger already do this ?

**Comment by maziac** [comment]
With "Disassembly at cursor" you get a disassembly at the cursor position. It will disassembly the whole selected area.
It's not at the PC position. So, this should work as you describe.

Furthermore there is the command "-dasm" that you can also use with labels, e.g.:
~~~
-dasm main 10
8000	DI 
8001	LD SP,80F1h 
8004	CALL 6047h 
8007	CALL 6052h 
800A	LD HL,6041h 
800D	LD (603Fh),HL 
8010	LD DE,5800h 
8013	LD HL,(603Fh) 
8016	LD A,(HL) 
8017	CALL 6017h 
~~~

And yes, the disassembly comes from memory. I.e. there is an own disassembler integrated.
It disassembles the opcode from memory. I.e. the user can check if the assembly code is still the right one, e.g. in case of self-modifying code.
BTW this is also true for the disassembly in the VARIABLES pane:
![Bildschirmfoto 2021-10-20 um 18 27 23](https://user-images.githubusercontent.com/3081973/138133143-11c19554-967a-4844-b44c-6a763fa58878.jpg)

(Note: this is for v2.4.0)

**Comment by infromthecold** [comment]
Awesome, thanks for that
